### PR TITLE
Update PDK, spec tests, and docs

### PR DESCRIPTION
**Pull request by genebean**
- Updated to PDK 1.9.1
  - Ran `pdk convert` to get this udpated to the current version of the PDK
  - Updated metadata.json to allow Puppet 6
  - Fixed open-ended module requirements per `pdk validate`
  - Fixed parameter ordering per `pdk validate`
  - Added `.sync.yml`
- Documentation and top-scope removal
  - Updated the readme to reference REFERENCE.md
  - Added badges to the readme
  - Moved documentaion from the readme to `init.pp`
  - Added basic docs to `windows_certificate.pp`
  - Generated REFERENCE.md
  - Removed top-scope from fact references and changed to structured facts
- Swtiched to facts from the puppet agent
- Add basic testing

### CODE REVIEW

**Comment by genebean** [comment]
@jonnytdevops I saw you had an open branch on this repo too... this PR includes a similar fix to what you had done but uses formal facts from the Puppet Agent instead. As a result, I just wanted to put this on your radar.

**Comment by genebean** [comment]
In contrast to #2, this PR does not make a parameter available to set the conf dir. It instead pulls it from the agent.

**Comment by genebean** [comment]
Also, I updated the labels in this repo to match what's expected by the changelog generator.

**Review by glennsarti — APPROVED** [review]
ship/cdn.vox-cdn.com/uploads/chorus_asset/file/15979212/Clippy_ShipIt.gif)

**Comment by jonnytdevops** [comment]
@glennsarti @genebean The fact used in this PR assumes the use of the puppet-agent module, which isn't something we're in a position to immediately deploy on user workstations. So that we can pin to a production SHA, I'd be grateful if another approach was used.

**Comment by genebean** [comment]
See notes in #3 regarding @jonnytdevops points.

File: REFERENCE.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+* [`windows_puppet_certificates::windows_certificate`](#windows_puppet_certificateswindows_certificate): 
+
+## Classes
+
+### windows_puppet_certificates
+
+This module takes the Puppet Master CA certificate and Puppet Agent client
+certificate and imports them into the Windows Certificate Store and marks the
+public key as Not Exportable. This is useful to allow Windows applications to
```
Comment (joshcooper) [inline REFERENCE.md:20]: private key?

Comment (glennsarti) [reply REFERENCE.md:20]: Whoops yes. typo.

Comment (genebean) [reply REFERENCE.md:20]: Merged #5 to fix this, thanks!

**Pull request merged**